### PR TITLE
[Bug] Fixed `LibreOffice::getText` method

### PR DESCRIPTION
**Pull request by Corepex**
## Error Description

It seems that the convert to txt (txt:Text) doesn't work properly ...
That effects certain command (e.g. `pimcore:search-backend-reindex`)

## Additional info  
fixed some logical issues in `LibreOffice::getText`.

### CODE REVIEW

File: lib/Document/Adapter/Ghostscript.php [hunk 1]
```diff
@@ -210,14 +210,18 @@ public function saveImage(string $imageTargetPath, $page = 1, $resolution = 200)
...
             if (!$asset && $this->asset) {
                 $asset = $this->asset;
             }
 
-            $path = $asset->getLocalFile();
+            if(!$path || !file_exists($path)) {
+                if(self::isFileTypeSupported($asset->getFilename())) {
+                    $path = $asset->getLocalFile();
+                }
+            }
```
Comment (brusch) [inline lib/Document/Adapter/Ghostscript.php:228]: @Corepex I guess we should throw an exception here in case `$path` is still null.

**Review by brusch — CHANGES_REQUESTED** [review]

File: lib/Document/Adapter/Ghostscript.php (hunk 1 above)
Comment (Corepex) [reply lib/Document/Adapter/Ghostscript.php:228]: done https://github.com/pimcore/pimcore/pull/14630/commits/9b91edc67c6d2a7166acca2d862a88318bda9c9b (bzw. https://github.com/pimcore/pimcore/pull/14630/commits/9fe406802dce6ce44c9b83b2941fd2c5b8b06a14)

File: lib/Document/Adapter/LibreOffice.php [hunk 2]
```diff
@@ -176,49 +176,23 @@ public function getPdf(?Asset\Document $asset = null)
...
-            $tmpName = PIMCORE_SYSTEM_TEMP_DIRECTORY . '/' . preg_replace("/\." . File::getFileExtension($localAssetTmpPath) . '$/', '.txt', $localAssetTmpPath);
-            if (file_exists($tmpName)) {
-                $text = file_get_contents($tmpName);
-                $text = \Pimcore\Tool\Text::convertToUTF8($text);
-                unlink($tmpName);
-
-                return $text;
+        try {
+            if (!parent::isFileTypeSupported($asset->getFilename())) {
+                $path = $this->getPdf($asset);
```
Comment (mcop1) [inline lib/Document/Adapter/LibreOffice.php]: `getPdf` method doesn´t return a `path`, but a `resource`. If the resource is passed to the parent´s getText method on line 190 an error will be thrown: 
grafik

File: lib/Document/Adapter/Ghostscript.php [hunk 3]
```diff
@@ -210,14 +210,22 @@ public function saveImage(string $imageTargetPath, $page = 1, $resolution = 200)
...
             }
 
-            $path = $asset->getLocalFile();
+            if(!$path || !file_exists($path)) {
+                if(self::isFileTypeSupported($asset->getFilename())) {
+                    $path = $asset->getLocalFile();
+                }
+
+                if(empty($path)) {
+                    throw new \Exception('Variable $path can\'t be null at this point.');
```
Comment (mcop1) [inline lib/Document/Adapter/Ghostscript.php]:
```suggestion
                    throw new \Exception('Could not get local file for asset with id ' . $asset->getId());
```
What do you think about adding a bit more detailed error message here?

File: lib/Document/Adapter/LibreOffice.php (hunk 2 above)
Comment (Corepex) [reply lib/Document/Adapter/LibreOffice.php]: thanks, seems like my cache didn't get cleared when using `bin/console c:c`. Fixed that within https://github.com/pimcore/pimcore/pull/14630/commits/ec8818703605158ee3d30fdfa3ee9f290550ecfd and https://github.com/pimcore/pimcore/pull/14630/commits/0d35077e610730078863ac0d1bed6610bb917c29

**Pull request merged**